### PR TITLE
Make sure the LedgerHandle close callback can be completed when encounter exception

### PR DESCRIPTION
**Pull request by codelipenghui**
Related to https://github.com/apache/pulsar/pull/12993

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java [hunk 1]
```diff
@@ -571,7 +571,11 @@ public void safeRun() {
 
                 // error out all pending adds during closing, the callbacks shouldn't be
                 // running under any bk locks.
-                errorOutPendingAdds(rc, pendingAdds);
+                try {
+                    errorOutPendingAdds(rc, pendingAdds);
+                } catch (Exception e) {
+                    closePromise.completeExceptionally(e);
```
Comment (zymap) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java:577]: If there has an exception, do we need to do the `tearDownWriteHandleState()` and ` MetadataUpdateLoop` operations?

Comment (hangc0276) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java:577]: Yes, it still needs the following operations. Catch this exception just to ensure the closePromise can complete.

**Review by hangc0276 — APPROVED** [review]

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java (hunk 1 above)
Comment (eolivelli) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java:577]: we shouldn't "completeExceptionally" or "complete" the `closePromise` twice

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java [hunk 2]
```diff
@@ -571,7 +571,11 @@ public void safeRun() {
 
                 // error out all pending adds during closing, the callbacks shouldn't be
                 // running under any bk locks.
-                errorOutPendingAdds(rc, pendingAdds);
+                try {
+                    errorOutPendingAdds(rc, pendingAdds);
+                } catch (Exception e) {
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java:576]: what about catching "Throwable" ?

**Review by eolivelli — CHANGES_REQUESTED** [review]

**Comment by codelipenghui** [comment]
@eolivelli Fixed, please check again.

**Review by StevenLuMT — APPROVED** [review]
lgtm

**Review by dlg99 — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
Lgtm

**Pull request merged**